### DISCUSSION TITLE
External temperature reading

### DISCUSSION THREAD

**Discussion by archaon** [revk/xESP32-Faikout#289] [General]
I am still really new to the Faikin modules (thank you for the most excellent 100% local control, btw), and I've been really struggling to understand how the Faikin can/will handle an external temperature reading, as well as the full extent of how Faikin Auto mode works.
My current understanding is that there is no way to override the indoor head unit's internal temperature sensor, even using a supported ble sensor.
Does pushing the room temperature via MQTT (for a non-ble sensor) only "do something" if Faikin Auto Mode is enabled?

**Comment by revk** [comment]
If you are sending only the temp via MQTT it will only do something if faikin auto mode is on.

If you also send the target via MQTT that takes over.

No it can't tell the Daikin the temp. It controls the set point high/low - the advanced section in the manual has more on how that works now.

**Reply by 44616E** [reply]
It will be interesting if some units do support it, for instance on cassette units the wire remote has the tempreature sensor so it must be sending it to the AC. But I expect most AC would not. Also the wire control is a different protocol.

**Reply by MartynasZilinskas** [reply]
Hey @revk, thank you for your amazing work. I have made a [virtual device project](https://github.com/MartynasZilinskas/mqtt-environment-monitor) and I am wondering if it's possible to see values that faikin sees from the MQTT commands? Like in BLE:
![image](https://github.com/user-attachments/assets/25b97eff-73af-4cc8-b318-f477e9bbab8b)

The app aggregates various mqtt temperature sensors in the room and sends average to faikin AC as control command:

Example payload to `command/faikin-ac/control`:
```json
{ 
    "env": 25.6, 
    "target": [25,27] 
}
```

**Reply by revk** [reply]
I am sure we have that in MQTT already!

**Reply by MartynasZilinskas** [reply]
![image](https://github.com/user-attachments/assets/c1a44b52-aad8-4151-9bff-020cb559a55f)
Yes, you're correct. Some why I was inspecting MQTT messages, not going directly to the web page.

**Reply by MartynasZilinskas** [reply]
With `tcontrol` we can set how long does the message is valid? Is the unit of the `tcontrol` value is seconds?

**Reply by revk** [reply]
Looks like seconds, yes

**Comment by snova69** [comment]
makes sense to me. thank you for your excellent work. The temperature setting in the A/C itself isn't really that important if the home automation code is the one responsible for telling the A/C to turn on or off. I'll just implement a negative delta in my home automation code to compensate to make sure the a/c doesn't stop cooling until the home automation system decides that is is cool enough.

**Comment by rug1024** [comment]
Do I understand this right: For Faikin Auto Mode I do not need a BLE temperature sensor when I
can supply room temperature via MQTT (e.g. from my ioBroker) to the Faikin device?

**Reply by revk** [reply]
You need to send the MQTT regularly, a `control` message, but correct. I do this with my own environmental monitoring sensors.

**Reply by 44616E** [reply]
@rug1024 you don't need an external temperature sensor at all for auto mode as it can use the AC temp. But it is an option if you want to.

**Reply by rug1024** [reply]
Understood. But I have an external Zigbee-Sensor which reports to my ioBroker, so I can send the value to the Faikin every minute.
Just could not figure out how exactly the message must be formatted. In the Faikin UI there only shows up "Env: ---".

**Reply by revk** [reply]
It is in the manual, honest, off the top of my head a `control` message with JSON payload like `{"env":22.5}` etc. But check the manual.

**Reply by rug1024** [reply]
maybe i forgot the {}. will try again. thanks!
